hera_pspec/grouping.py

```diff
@@ -386,6 +405,14 @@ def average_spectra(uvp_in, blpair_groups=None, time_avg=False,
                     wgts = uvp.get_wgts(spw, blp, p)
                     ints = uvp.get_integrations(spw, blp, p)[:, None]
                     w = (ints * np.sqrt(nsmp))
+
+                    # Get error bar weights and set invalid ones to zero.
+                    errws = {}
+                    for stat in stat_l:
+                        errs = uvp.get_stats(stat, (spw, blp, p))
+                        errs[np.where(errs > 0.)] = 1. / errs[np.where(errs > 0.)] ** 2.
```

Comment:
lot of repeated indexing here, you should do the indexing once, store it and then reference it later on. something like
```
for stat in stat_l:
    errw = uvp.get_stats(stat, (spw, blp, p))
    errw_bools = errs > 0.0
    errw[errw_bools] = 1.0 / errw[errw_bools] ** 2.0
    errw[~errw_bools] = 0.0
    errws[stat] = errw
```